conf/broker.conf

```diff
@@ -567,6 +567,9 @@ exposeTopicLevelMetricsInPrometheus=true
 # Enable consumer level metrics. default is false
 exposeConsumerLevelMetricsInPrometheus=false
 
+# Classname of Pluggable JVM GC metrics logger that can log GC specific metrics
+jvmGCMetricsLoggerClassName=java.lang.management.ManagementFactory.JvmG1GCMetricsLogger
```

Comment:
so why is the default "JvmG1GCMetricsLogger"? and not "JvmDefaultGCMetricsLogger.java"?

Reply:
right now, default implementation is G1GC and if we change it then it will be breaking change for users with existing configuration. We had similar discussion on log4j configuration change in past so, we decided to maintain compatibility with previous release.

Reply:
I have made change for auto-detection so, if user is using G1GC then it will auto select G1GC impl else it will always fall back to default implementation.